### ISSUE TITLE
consistent-return: wrong reporting location

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#14788]
**Tell us about your environment**

* **ESLint Version:** 7.30.0
* **Node Version:** v14
* **npm Version:** v7
* **Operating System:**  macOS

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
/*eslint consistent-return: "error"*/
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function name(condition) { 
  if (condition) {
    return;
  }
  else {
    return false;
  }
}
```

**What did you expect to happen?**
To report the return inside the if block.

```js
function name(condition) { 
  if (condition) {
    return;
//  ^^^^^^^
  }
  else {
    return false;
  }
}
```

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

reporting the `return` in `else` block
```js
function name(condition) { 
  if (condition) {
    return;
  }
  else {
    return false;
//  ^^^^^^^^^^^^^
  }
}
```

**Steps to reproduce this issue:**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgY29uc2lzdGVudC1yZXR1cm46IFwiZXJyb3JcIiovXG5cblxuZnVuY3Rpb24gbmFtZShjb25kaXRpb24pIHsgXG4gIGlmIChjb25kaXRpb24pIHtcbiAgICByZXR1cm47XG4gIH1cbiAgZWxzZSB7XG4gICAgcmV0dXJuIGZhbHNlO1xuICB9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Are you willing to submit a pull request to fix this bug?**

Yes, I will fix it if accepted.

**Comment by nzakas** [comment]
I believe this is working as intended. The rule flags the first return statement that is inconsistent with the previous return statements. So, we examine the first return statement and say that represents what return statements in this function should be. Then we look at the second statement and see that it doesn’t match the first, so we flag the second as inconsistent.

**Comment by snitin315** [comment]
I see, that makes sense.
